src/pages/CustomerInvoicesPage.js

```diff
@@ -3,252 +3,231 @@
-    this.dataFilters = {
-      searchTerm: '',
-      sortBy: 'serialNumber',
-      isAscending: false,
-    };
-  }
-
-  onNewInvoice = otherParty => {
-    const { database, currentUser } = this.props;
-    let invoice;
-    database.write(() => {
-      invoice = createRecord(database, 'CustomerInvoice', otherParty, currentUser);
-    });
-    this.navigateToInvoice(invoice);
+import {
+  PageButton,
+  CheckedComponent,
+  UncheckedComponent,
+  DisabledCheckedComponent,
+  DisabledUncheckedComponent,
+} from '../widgets';
+import { DataTable, Row, Cell, CheckableCell, DataTableHeaderRow } from '../widgets/DataTable';
+import {
+  selectRow,
+  deselectRow,
+  deselectAll,
+  sortData,
+  filterData,
+  openBasicModal,
+  closeBasicModal,
+  deleteTransactionsById,
+} from './dataTableUtilities/actions';
+import { newSortDataBy } from '../utilities';
+import usePageReducer from '../hooks/usePageReducer';
+import DataTablePageView from './containers/DataTablePageView';
+
+import { SUSSOL_ORANGE, newDataTableStyles, newPageStyles } from '../globalStyles';
+
+const keyExtractor = item => item.id;
+
+const initialState = () => {
+  const backingData = UIDatabase.objects('CustomerInvoice');
+  return {
+    backingData,
+    data: newSortDataBy(backingData.slice(), 'serialNumber', false),
+    keyExtractor,
+    dataState: new Map(),
+    searchTerm: '',
+    filterDataKeys: ['otherParty.name'],
+    sortBy: 'serialNumber',
+    isAscending: false,
+    modalKey: '',
+    hasSelection: false,
   };
+};
 
-  onDeleteConfirm = () => {
-    const { selection, transactions } = this.state;
-    const { database } = this.props;
-
-    database.write(() => {
-      const transactionsToDelete = [];
-      for (let i = 0; i < selection.length; i += 1) {
-        const transaction = transactions.find(
-          currentTransaction => currentTransaction.id === selection[i]
-        );
-        if (transaction.isValid() && !transaction.isFinalised) {
-          transactionsToDelete.push(transaction);
-        }
-      }
-      database.delete('Transaction', transactionsToDelete);
-    });
-    this.setState({ selection: [] });
-    this.refreshData();
-  };
-
-  onDeleteCancel = () => {
-    this.setState({ selection: [] });
-    this.refreshData();
-  };
-
-  onRowPress = invoice => this.navigateToInvoice(invoice);
-
-  onSelectionChange = newSelection => this.setState({ selection: newSelection });
+export const CustomerInvoicesPage = ({ currentUser, navigateTo, routeName }) => {
+  const [state, dispatch, instantDebouncedDispatch, debouncedDispatch] = usePageReducer(
+    routeName,
+    initialState()
+  );
+  const { data, dataState, sortBy, isAscending, columns, modalKey, hasSelection } = state;
 
-  navigateToInvoice = invoice => {
-    const { database, navigateTo } = this.props;
+  const renderNewInvoiceButton = () => (
+    <PageButton
+      text={buttonStrings.new_invoice}
+      onPress={() => dispatch(openBasicModal('Anything will do'))}
```

Comment:
accidental param?

Reply:
Lol that was a joke, but technically the statement in that string is true.

Should use constant you reckon?

Reply:
I guess it's fine - there is a PR soon for factoring modal logic. If you like that it might change how this is done, but since this page only uses one modal, maybe not. Maybe just confusing for the next person who looks at this code, and a little annoying for the person who has to add an extra modal to this page

Reply:
https://github.com/openmsupply/mobile/pull/1164/commits/07bf3098cfa0aabd16e0fa37ef9005ad4cc004d2